index/scorch/introducer.go

```diff
@@ -241,3 +252,48 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 	// notify merger we incorporated this
 	close(nextMerge.notify)
 }
+
+func (s *Scorch) revertToSnapshot(revertTo *snapshotReversion) error {
+	defer close(revertTo.applied)
+
+	if revertTo.snapshot == nil {
+		return fmt.Errorf("Cannot revert to a nil snapshot")
```

Comment:
need to also send this error onto the revertTo.applied channel?

Reply:
Done.